### ISSUE TITLE
508-defect-1 ❗ Launchblocker [AXE-CORE, SCREENREADER]: IDs of active elements MUST be unique

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#12674]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

This review screen document has active elements with the same id attribute: `root_serviceInformation__title`. IDs of active elements **must** be unique.

### Why it matters

The ID attribute uniquely identifies focusable elements on a page. It does not make sense to duplicate an active ID.

Duplicate active ID values break the accessibility of focusable elements including labels for forms, table header cells, etc., Screen readers and client-side scripts will skip any duplication other than the first instance. Validating HTML files help prevent and eliminate possible sources of accessibility problems, when not breaking accessibility.

Those experienced with client-side scripting know that when you re-use an active ID, typically the only one that gets acted upon by the scripting is the first instance of the use of that active ID. Similarly, assistive technologies may, when referencing an active ID, only reference the first one accurately.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As an assistive tech user, I want to navigate the document without any conflicts of data structure that makes it unusable.

## Environment

* Operating System: any
* Browser: any
* Server destination: staging

## Steps to Recreate

1. Logged in as user 261, enter https://staging.va.gov/disability/file-disability-claim-form-21-526ez/review-and-submit in browser
2. This should be on the Review screen
3. Open all of the accordion sections and editable areas
4. Using developer tools, run an axe scan
5. Verify that "IDs of active elements must be unique" appears as an issue, related to:

```
Element location
#collapsible-1 > .schemaform-chapter-accordion-content.usa-accordion-content[aria-hidden="false"] > .form-review-panel-page:nth-child(5) > .rjsf[novalidate=""] > fieldset > .input-section.schemaform-field-container > .schemaform-block-header > .schemaform-block-subtitle.schemaform-block-title
Element source
<legend class="schemaform-block-title schemaform-block-subtitle" id="root_serviceInformation__title" tabindex="-1">Reserves and National Guard Information</legend>
```

## Possible Fixes (optional)

```diff
#collapsible-3 > .schemaform-chapter-accordion-content.usa-accordion-content[aria-hidden="false"] > .form-review-panel-page:nth-child(5) > .rjsf[novalidate=""] > fieldset > .input-section.schemaform-field-container > .schemaform-block-header > .schemaform-block-subtitle.schemaform-block-title
Element source
<legend 
  class="schemaform-block-title schemaform-block-subtitle" 
!  id="root_serviceInformation__title" 
+  id="root_serviceInformation__title_unique" 
  tabindex="-1">Reserves and National Guard Information</legend>
```

## WCAG or Vendor Guidance (optional)

* [Deque University - ID attribute value must be unique](https://dequeuniversity.com/rules/axe/3.5/duplicate-id-active) WCAG 2.0 (A): MUST

* [W3C Understanding WCAG 2.0 - 4.1.1 Parsing](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-parses.html)

## Screenshots or Trace Logs

![Screen Shot 2020-08-20 at 1 51 09 PM](https://user-images.githubusercontent.com/57469/90807272-39512580-e2ec-11ea-8d1c-f2dcc31fecec.png)

**Comment by jenstrickland** [comment]
@1Copenut Should this remain a 508-defect-1 ❗ Launchblocker — as this is platform code that needs to be fixed?

cc @Mottie @luke-gcio

**Comment by 1Copenut** [comment]
@billfienberg just opened a forms-library issue for this: https://github.com/department-of-veterans-affairs/va.gov-team/issues/12747

Adding comment here to link the two issues.

**Comment by luke-gcio** [comment]
We will leave this open for tracking of the original work but reducing estimate to 1 since we are not doing the work.  Removing from sprint.

**Comment by Mottie** [comment]
We need to look at this again. VSP merged in a fix and closed #12747.

**Comment by Mottie** [comment]
@billfienberg I just tested this using the latest master, and it still appears to be a problem.

![Screen Shot 2020-09-11 at 3 37 48 PM](https://user-images.githubusercontent.com/136959/92970742-c8adac80-f444-11ea-92be-9f4af54748a5.png)

**Comment by jenstrickland** [comment]
@Mottie — is this a sitewide issue?

**Comment by Mottie** [comment]
Fixed in https://github.com/department-of-veterans-affairs/vets-website/pull/17255

**Comment by 1Copenut** [comment]
I ran through the app listed in the Description. 0 axe-core violations. Moving to closed!
